src/circuit/circuit_frame.rs

```diff
@@ -3089,9 +3118,9 @@ mod tests {
             assert!(delta == Delta::Equal);
 
             //println!("{}", print_cs(&cs));
-            assert_eq!(32114, cs.num_constraints());
+            assert_eq!(32144, cs.num_constraints());
```

Comment:
This is good to see and seems to confirm the design has relatively small marginal cost.